premium.sol

```diff
@@ -241,7 +242,7 @@ contract premium is module, safeMath {
      * @param _extraData Data to give forward to the receiver
      * @return Whether the transfer was successful or not
      */
-    function transferAndCall(address _to, uint256 _amount, bytes _extraData) external returns (bool) {
+    function transfer(address _to, uint256 _amount, bytes _extraData) external returns (bool) {
```

Comment:
Function `transfer` should filter if the receiver is a contract or an address. If the receiver is a contract the `receiveCorionToken` function should be called at receiver. If the receiver is an externally owned account (address) nothing should be called.
If you will try to execute `transferToContract` to transfer to address then the transaction will always fail.